components/percona_telemetry/logger.h

```diff
@@ -0,0 +1,31 @@
+#ifndef PERCONA_TELEMETRY_LOGGER_H
+#define PERCONA_TELEMETRY_LOGGER_H
+
+#include <mysql/components/component_implementation.h>
+#include "common.h"
+
+#define LOG_COMPONENT_TAG CURRENT_COMPONENT_NAME_STR
+#include <mysql/components/services/log_builtins.h>
+
+class Logger {
+ public:
+  Logger(SERVICE_TYPE(log_builtins) & log_builtins_service,
+         SERVICE_TYPE(log_builtins_string) & log_builtins_string_service,
+         loglevel log_level = SYSTEM_LEVEL);
+
+  ~Logger() = default;
+
+  Logger(const Logger &rhs) = delete;
+  Logger(Logger &&rhs) = delete;
+  Logger &operator=(const Logger &rhs) = delete;
+  Logger &operator=(Logger &&rhs) = delete;
+
+  void info(const char *format, ...) __attribute__((format(printf, 2, 3)));
```

Comment:
I believe there is a wrapper macro for `__attribute__ `.
Probably `MY_ATTRIBUTE`

Reply:
fixed